### PR TITLE
Merge TLP armors and make TLP weapons into prototypes

### PR DESCRIPTION
**Pull request by cannonfodder1**
#130

### CODE REVIEW

**Comment by cannonfodder1** [comment]
> Can you explain the reasoning behind these changes?

So if you have a item with a required tech, then after the tech is finished a pop up will come up with the weapon’s loadout image, the blue one with all the modular parts missing. It looks horrible. So instead the items have a required schematic, and the schematic is requiring/granted by the tech. This way, the pop up will show the weapon’s schematic image, with full colour and no modular parts removed.

**Comment by Xymanek** [comment]
I see why you did it, but I do not think that involving **vanilla** items into our TLE shenanigans is a good approach. Are you sure that there is no other way (that doesn't involve **vanilla** items)?

**Comment by cannonfodder1** [comment]
Well, I'm kind of going outside the scope of this PR here. The issues with the popups looking like crap result from our singlebuild system, which, I will note, already touches the vanilla items. If we decide that this cure is worse than the disease, then it would be perfectly fine to just leave it as is and nothing functional would change. It's only an aesthetic problem.

**Comment by Xymanek** [comment]
> The issues with the popups looking like crap result from our singlebuild system, which, I will note, already touches the vanilla items

Wait, are vanilla items looking bad from single build changes?

**Comment by cannonfodder1** [comment]
> Wait, are vanilla items looking bad from single build changes?

Yeah, I tested on the master branch and the broken images happen there. The problem is that the tech popup shows the item's image, not the schematic's image, because we cut the schematic out of the equation. If we change the item's image, then weapon upgrades will break the image when you look at it in the loadout. If we do nothing, the image is broken when you look at it in the tech popup. The only solution I can see is to bring back the schematic into the system, invisible to the player and disappearing once the tech is researched.

**Comment by Xymanek** [comment]
Right, I see (and confirm) the issue. Sucks that it wasn't brought to attention before, but whatever.

> Well, I'm kind of going outside the scope of this PR here

Yeah, don't. This solution is ugly and I don't think the issue is worth delaying alpha 2. Just leave it as-is now and I'll think about potential solutions

File: CovertInfiltration/Src/CovertInfiltration/Classes/X2StrategyElement_InfiltrationActions.uc [hunk 1]
```diff
@@ -39,12 +39,12 @@ static function X2CovertInfiltrationTemplate CreateInfiltrationTemplate(name Cov
 
 	if (bCreateSlots)
 	{
+		Template.Slots.AddItem(CreateDefaultOptionalSlot('InfiltrationStaffSlot'));
```
Comment (Xymanek) [inline CovertInfiltration/Src/CovertInfiltration/Classes/X2StrategyElement_InfiltrationActions.uc]: What's the point of this change?

Comment (cannonfodder1) [reply CovertInfiltration/Src/CovertInfiltration/Classes/X2StrategyElement_InfiltrationActions.uc]: Oh, whoops. That probably shouldn't be in this PR. Basically it just moves the optional slots to be on either side with the required four in the middle, like vanilla. How it was setup before, both optional slots were together on the right side.

File: CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration_TemplateMod.uc [hunk 2]
```diff
@@ -162,6 +273,70 @@ static function DisableLockAndBreakthrough()
...
+			if(TechTemplate.ResearchCompletedFn == none)
+			{
+				TechTemplate.ResearchCompletedFn = GiveAllItems;
+			}
+			TechTemplate.ItemRewards.AddItem(Prototype);
+		}
+	}
+}
+
+static function GiveAllItems(XComGameState NewGameState, XComGameState_Tech TechState)
```
Comment (Xymanek) [inline CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration_TemplateMod.uc]: Why do we need both this and the event listener?

File: CovertInfiltration/Src/CovertInfiltration/Classes/X2EventListener_Infiltration.uc [hunk 3]
```diff
@@ -234,6 +235,35 @@ static protected function EventListenerReturn ShouldCleanupCovertAction(Object E
...
+	ItemTemplateManager = class'X2ItemTemplateManager'.static.GetItemTemplateManager();
+	
+	TechState = XComGameState_Tech(EventData);
+
+	if(TechState == none) return ELR_NoInterrupt;
+
+	ItemRewards = TechState.GetMyTemplate().ItemRewards;
+	foreach ItemRewards(ItemName)
+	{
+		`LOG("Evaluating Item:" @ string(ItemName));
```
Comment (Xymanek) [inline CovertInfiltration/Src/CovertInfiltration/Classes/X2EventListener_Infiltration.uc]: From `README.md`:

> Use custom log macros - `CI_Log`, `CI_Warn` and `CI_Trace`. Any PR that uses `ScriptLog` (eg. `log("Something")`) __will not be merged__

**Comment by Xymanek** [comment]
So i've tested it with mag weapons and it seems to work

File: CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration_TemplateMod.uc (hunk 2 above)
Comment (cannonfodder1) [reply CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration_TemplateMod.uc]: Tried to call XComHQ's GiveItem function from inside the EventListenerReturn, but it didn't work with that function's GameState variable. So I went back here, where the OnResearchedFn delegate function has a NewGameState that can properly call GiveItem.

File: CovertInfiltration/Src/CovertInfiltration/Classes/X2EventListener_Infiltration.uc (hunk 3 above)
Comment (cannonfodder1) [reply CovertInfiltration/Src/CovertInfiltration/Classes/X2EventListener_Infiltration.uc]: Don't really need that log anymore, was for debugging the issue with armor prototypes. I'll remove it

File: CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration_TemplateMod.uc (hunk 2 above)
Comment (Xymanek) [reply CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration_TemplateMod.uc]: It didn't wok as the passed gamestate is already submitted. You need to make a new one

**Comment by Xymanek** [comment]
Please remove the commented out code. Also, the log message is still there

**Pull request merged**